What are you trying to achieve?
Think of the problem and the data, check the description and get familiarized to the domain. 

This would be the time you look at the data description, check what columns you have, what types of values.

In [5]:
import pandas as pd
import numpy as np


#import dataframe from csv file
df = pd.read_csv('data/eda_housing_market.csv')

In [11]:
#show first 5 rows of dataframe
df.head()

house_id        date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  2014-10-13  221900.0       3.0       1.00       1180.0   
1  6414100192  2014-12-09  538000.0       3.0       2.25       2570.0   
2  5631500400  2015-02-25  180000.0       2.0       1.00        770.0   
3  2487200875  2014-12-09  604000.0       4.0       3.00       1960.0   
4  1954400510  2015-02-18  510000.0       3.0       2.00       1680.0   

   sqft_lot  floors  waterfront  view  ...  sqft_basement  yr_built  \
0    5650.0     1.0         NaN   0.0  ...            0.0      1955   
1    7242.0     2.0         0.0   0.0  ...          400.0      1951   
2   10000.0     1.0         0.0   0.0  ...            0.0      1933   
3    5000.0     1.0         0.0   0.0  ...          910.0      1965   
4    8080.0     1.0         0.0   0.0  ...            0.0      1987   

   yr_renovated  zipcode      lat     long  sqft_living15  sqft_lot15  \
0           0.0    98178  47.5112 -122.257         1340.0      5650.0   
1       19910.0    98125  47.7210 -122.319         1690.0      7639.0   
2           NaN    98028  47.7379 -122.233         2720.0      8062.0   
3           0.0    98136  47.5208 -122.393         1360.0      5000.0   
4           0.0    98074  47.6168 -122.045         1800.0      7503.0   

   year_sold  age  
0       2014   59  
1       2014   63  
2       2015   82  
3       2014   49  
4       2015   28  

[5 rows x 23 columns]

In [7]:
# checking the datatype and number of nulls and rows
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   house_id       21597 non-null  int64  
 1   date           21597 non-null  object 
 2   price          21597 non-null  float64
 3   bedrooms       21597 non-null  float64
 4   bathrooms      21597 non-null  float64
 5   sqft_living    21597 non-null  float64
 6   sqft_lot       21597 non-null  float64
 7   floors         21597 non-null  float64
 8   waterfront     19206 non-null  float64
 9   view           21534 non-null  float64
 10  condition      21597 non-null  int64  
 11  grade          21597 non-null  int64  
 12  sqft_above     21597 non-null  float64
 13  sqft_basement  21145 non-null  float64
 14  yr_built       21597 non-null  int64  
 15  yr_renovated   17749 non-null  float64
 16  zipcode        21597 non-null  int64  
 17  lat            21597 non-null  float64
 18  long  

The data is clean and no nulls exists in the dataset. It has 21597 rows of house price data. Except of the date, all columns are numerical

In [10]:
# extract the year in order to calcuate the age of the house.
df['year_sold'] = pd.to_datetime(df['date']).dt.year
df['age'] = df['year_sold'] - df['yr_built']

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 23 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   house_id       21597 non-null  int64  
 1   date           21597 non-null  object 
 2   price          21597 non-null  float64
 3   bedrooms       21597 non-null  float64
 4   bathrooms      21597 non-null  float64
 5   sqft_living    21597 non-null  float64
 6   sqft_lot       21597 non-null  float64
 7   floors         21597 non-null  float64
 8   waterfront     19206 non-null  float64
 9   view           21534 non-null  float64
 10  condition      21597 non-null  int64  
 11  grade          21597 non-null  int64  
 12  sqft_above     21597 non-null  float64
 13  sqft_basement  21145 non-null  float64
 14  yr_built       21597 non-null  int64  
 15  yr_renovated   17749 non-null  float64
 16  zipcode        21597 non-null  int64  
 17  lat            21597 non-null  float64
 18  long  